src/group_theory/amenable/def_amenable.lean

```diff
@@ -0,0 +1,169 @@
+/-
+Copyright (c) 2022 Matthias Uschold. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Matthias Uschold.
+-/
+
+import .mean
+
+/-!
+# Amenable Groups
+
+We introduce the concept of amenable groups.
+A (topological) group is called `amenable` if there exists a left-invariant mean,
+i.e. a mean (a positive, normalised linear function bounded_continuous_function G ℝ → ℝ )
+that is invariant under left translation of the argument.
+
+
+## Main Definitions
+
+- `left_invariant_mean`  : mean with the property of being left-invariant
+- `right_invariant_mean` : Alternative variant
+- `bi_invariant_mean`    : Alternative variant
+- `amenable`             : amenability of a group (defined by nonemptiness of left_invariant_mean)
+
+
+We will see in `right_bi_invariance.lean` that the alternative variants yield
+the same notion of amenability
+
+## Implementation Notes
+
+The notion of a mean is already defined in the file `mean.lean`. This
+file defines the left translate of a function and uses this to define
+amenability.
+
+## References
+* [C. Löh, *Geometric Group Theory*, Definition 9.1.1][loeh17]
+* <https://en.wikipedia.org/wiki/Amenable_group>
+* [A.L.T. Paterson, *Amenability*, Definition 0.2][Paterson1988]
+
+## Tags
+
+amenable, amenability, invariant mean, left invariant mean
+-/
+
+open mean
+open classical
+open bounded_continuous_function
+
+variables {G:Type*} [uniform_space G] [group G] [topological_group G]
+
+
+/-- left-translate of a function by translating the argument -/
+def left_translate (g : G) (f : bounded_continuous_function G ℝ) :
+  bounded_continuous_function G ℝ :=
+f.comp_continuous (continuous_map.mk (λ h, g⁻¹*h) (continuous_mul_left g⁻¹))
+
+
+
+@[simp]
+lemma left_translate_eval {g : G} {f : bounded_continuous_function G ℝ} :
+  ∀(x:G), left_translate g f x = f(g⁻¹*x) := by tauto
+
+
+
+instance left_translate_action  : mul_action G (bounded_continuous_function G ℝ) :=
+@mul_action.mk G  (bounded_continuous_function G ℝ) _
```

Comment:
The preferred way to spell this is by using `{ smul := _, ... }` notation

Reply:
done